[mfjkri] iP

mfjkri opened the pull request:
# Chatty

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Chatty frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ _SUPER FAST_ to use

All you need to do is,

1. download it from [here](https://github.com/nus-cs2103-AY2324S1/ip/pull/25).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE!**

Features:

- [x]  Managing tasks
- [ ]  Managing deadlines (coming soon)
- [ ]  Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

mfjkri committed:
Add Level-0

mfjkri committed:
Add Level-1

mfjkri committed:
Add Level-2

mfjkri committed:
Add Level-3

mfjkri committed:
Add Level-4

mfjkri committed:
A-TextUiTesting

mfjkri committed:
Add Level-5

mfjkri committed:
Add Level-6

mfjkri committed:
Update exception handling

mfjkri committed:
Add Level 7 functionality

mfjkri committed:
Fix saving and loading

mfjkri committed:
Merge branch 'branch-Level-7'

mfjkri committed:
Add Level-8

mfjkri committed:
Merge branch 'branch-Level-8'

mfjkri committed:
Add time support

mfjkri committed:
Add Level A More OOP

mfjkri committed:
Merge branch 'branch-A-MoreOOP'

mfjkri committed:
Add Level A Packages

mfjkri committed:
Merge branch 'branch-A-Packages'

mfjkri committed:
Merge branch 'add-gradle-support' into branch-A-Gradle

mfjkri committed:
Add Level A Gradle

mfjkri committed:
Merge branch 'branch-A-Gradle'

mfjkri committed:
Add Level A JUnit

mfjkri committed:
Add Level A Javadoc

mfjkri committed:
Merge branch 'branch-A-Javadoc'

mfjkri committed:
Add Level 9

mfjkri committed:
Merge branch 'Level-9'

mfjkri committed:
Fix checkstyle

mfjkri committed:
Merge branch 'branch-A-CheckStyle'

mfjkri committed:
Add Level 10

mfjkri committed:
Merge branch 'branch-Level-10'

tanteckfang commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,72 @@
+        return ui.getStatus();
+    }
+
+    public boolean getExitStatus() {
```
Good method's name is used

tanteckfang commented on src/main/java/duke/lib/Storage.java:
```diff
@@ -0,0 +1,90 @@
+     * @param directoryName Directory where the file will be stored.
+     * @param fileName      Name of the save file.
+     */
+    public Storage(String directoryName, String fileName) {
```
good use of camel case

tanteckfang commented on src/main/java/duke/lib/Storage.java:
```diff
@@ -0,0 +1,90 @@
+     *
+     * @throws DukeException If an unexpected error occurs while creating the directory or file.
+     */
+    private void createIfMissing() throws DukeException {
```
alternative can be createMissingFile()

tanteckfang commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,46 @@
+    /**
+     * Due date of the deadline task.
+     */
+    protected LocalDateTime by;
```
by does not provide much context, perhaps u can use dueDate.

tanteckfang commented on src/main/java/duke/ui/MainWindow.java:
```diff
@@ -0,0 +1,85 @@
+ */
+public class MainWindow extends AnchorPane {
+    @FXML
+    private ScrollPane scrollPane;
```
👍 class variable is private

tanteckfang commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,72 @@
+package duke;
+
+import duke.command.Command;
```
👍 Imported classes listed explicitly.

Kurtyjlee commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,72 @@
+ */
+public class Duke {
+
+    private final Storage storage;
```
Good variable names

Kurtyjlee commented on src/main/java/duke/command/DeleteCommand.java:
```diff
@@ -0,0 +1,39 @@
+ * Represents a command to delete a task from the task list.
+ */
+public class DeleteCommand extends Command {
+    private final int index;
```
Might want to be more specific with the naming here

Kurtyjlee commented on src/main/java/duke/command/FindCommand.java:
```diff
@@ -0,0 +1,33 @@
+ * Represents a command to search for tasks containing a specified keyword.
+ */
+public class FindCommand extends Command {
+    private String searchKeyword;
```
Might want to choose a more intentioning naming here

Kurtyjlee commented on src/main/java/duke/command/MarkCommand.java:
```diff
@@ -0,0 +1,36 @@
+/**
+ * Represents a command to mark a task as done in the task list.
+ */
+public class MarkCommand extends Command {
```
Good use to separate classes for different commands

Kurtyjlee reviewed (approved):
Not many comments overall as this was very well done. Just a few nits here and there for you to consider.

mfjkri committed:
Add assertions for UI elements

Assumptions is made about UI elements being present at runtime.

However this might not always be the case and missing UI elements might lead to unexpected behavior.

Assertions let us catch these missing UI elements and stop execution.

Assertions can easily be disabled for production and ensure correct functionality where expected.

Refer to this for assertions reference:
https://nus-cs2103-ay2324s1.github.io/website/se-book-adapted/chapters/errorHandling.html#assertions

mfjkri committed:
Add missing javadocs

Missing javadocs can hurt the readability of the code for future readers
as they might be unable to understand what the function does or requires.

Adding missing headers to all public classes and methods helps us
to increase code readiblity and maintainability.

mfjkri committed:
Merge pull request #1 from mfjkri/branch-A-Assertions

Add assertions for UI elements

mfjkri committed:
Merge pull request #2 from mfjkri/branch-A-CodeQuality

Add missing javadocs

mfjkri committed:
Add CI pipeline

mfjkri committed:
Merge branch 'branch-A-CI'

mfjkri committed:
Add friendlier syntax and remove parameter order

Currently parameters must be specified in a specific order and
the root command can be long to type.

This makes it troublesome of the users to remember the order of parameters
as well as the full word of the command.

We now allow for parameters to be specified in any order while still
having proper input validation for existence and typing of it.
Commands are also given shortened forms as shortcut (e.g. delete: rm).

This way we allow for greater flexibility and make the CLI more intuitive to
the user.

mfjkri committed:
Merge branch 'branch-C-FriendlierSyntax'

mfjkri committed:
Fix check style errors

mfjkri committed:
Fix code quality

mfjkri committed:
Add User Guide

mfjkri committed:
Merge branch 'branch-A-UserGuide'

mfjkri committed:
Update User Guide

mfjkri committed:
Fix javadocs format
